drizzlepac/alignimages.py

```diff
@@ -0,0 +1,1075 @@
+                                item.best_meta = item.meta.copy()
+                            best_fitStatusDict = fitStatusDict.copy()
+                            best_fitQual = fitQual
+                        elif fitQual == best_fitQual: # new solution same level of fitQual. Choose whichever one has the lowest total rms as "best" and keep looping.
+                            if best_fit_rms >= 0.:
+                                if fit_rms < best_fit_rms:
+                                    best_fit_rms = fit_rms
+                                    best_fit_num = fit_num
+                                    for item in imglist:
+                                        item.best_meta = item.meta.copy()
+                                    best_fitStatusDict = fitStatusDict.copy()
+                        else: # new solution has worse fitQual. discard and continue looping.
+                            continue
+                except Exception:
+                    print("WARNING: Catastrophic fitting failure with catalog {} and matching algorithm {}.".format(catalogList[catalogIndex],algorithm_name.__name__))
+                    filteredTable['status'][:] = 1
+                    filteredTable['processMsg'][:] = "Fitting failure"
+                    # It may be there are additional catalogs and algorithms to try, so keep going
+                    continue
+
+                if fitQual == 1:  # break out of inner fit algorithm loop
+                    break
+        if fitQual == 1: #break out of outer astrometric catalog loop
+            break
+    currentDT = datetime.datetime.now()
+    deltaDT = (currentDT - startingDT).total_seconds()
+    print('Processing time of [STEP 5b]: {} sec'.format(deltaDT))
+    startingDT = currentDT
+    # 6: Populate the filteredTable
+    print("-------------------- STEP 6: Collect up information and populate the filtered table --------------------")
+    if best_fit_rms > 0 and best_fit_rms < MAX_FIT_RMS:
+        print("The fitting process was successful with a best fit total rms of {} mas".format(best_fit_rms))
+    else:
+        print("The fitting process was unsuccessful with a best fit total rms of {} mas".format(best_fit_rms))
+
+    if best_fit_rms > 0 and best_fit_rms < MAX_FIT_LIMIT:
+        # update to the meta information with the lowest rms if it is reasonable
+        for item in imglist:
+            item.meta = item.best_meta.copy()
+        filteredTable['status'][:] = 0
+        fitStatusDict = best_fitStatusDict.copy()
+
+        # Protect the writing of the table within the best_fit_rms
+        info_keys = OrderedDict(imglist[0].meta['tweakwcs_info']).keys()
+        # Update filtered table with number of matched sources and other information
+        for item in imglist:
+            imgname = item.meta['name']
+            index = np.where(filteredTable['imageName'] == imgname)[0][0]
+
+            if item.meta['tweakwcs_info']['status'].startswith("FAILED") != True:
```

Comment:
@mdlpstsci This should be written as `if not ...starswith('FAILED')`. `starswith` returns a boolean result.